src/mixins/embed.js

```diff
@@ -506,5 +515,25 @@ export default {
       }
       // TO-DO: handle go to portfolio
     },
+    async updateTempCTA() {
+      this.tempLink = `https://app.rinkeby.like.co/nfttest/fetch?url=${this.referrer}&liker_id=${this.id}`;
+      this.tempCTA = 'Create ISCN';
+      this.tempNFTPrice = -1;
+      if (this.iscnId) {
+        const res = await axios.get(`${LIKECOIN_API}/likernft/mint`, {
+          params: {
+            iscn_id: this.iscnId,
+          },
+        }).catch(() => { });
+        if (res) {
+          this.tempLink = `https://app.rinkeby.like.co/nfttest/button/${encodeURIComponent(this.iscnId)}`;
+          this.tempCTA = 'Collect NFT';
+          this.tempNFTPrice = res.data.currentPrice;
+        } else {
+          this.tempLink = `https://app.rinkeby.like.co/nfttest/mint/${encodeURIComponent(this.iscnId)}`;
+          this.tempCTA = 'Mint NFT';
```

Comment:
These can be computed in view components, the only data we need to pass is `NFTCurrentPrice`